\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={R-code for Heterozygosity at neutral and immune loci does not influence neonatal mortality due to microbial infection in Antarctic fur seals},
            pdfauthor={Vivienne Litzke, Meinolf Ottensmann, Jaume Forcada \& Joseph I. Hoffman},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{R-code for `Heterozygosity at neutral and immune loci does not influence
neonatal mortality due to microbial infection in Antarctic fur seals'}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Vivienne Litzke, Meinolf Ottensmann, Jaume Forcada \& Joseph I. Hoffman}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
    \date{}
    \predate{}\postdate{}
  
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage[table]{xcolor}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}

\begin{document}
\maketitle

\subsubsection{Preface}\label{preface}

This document provides all the \texttt{R\ code} used for our paper. Both
the Rmarkdown file and the data can be downloaded from the accompanying
GitHub repository on (URL TO GITHUB) as a zip archive containing all the
files. Our data originates from samples collected from a colony of
Antarctic fur seals (\emph{Arctocephalus gazella}) at Bird Island, South
Georgia between the years of 2000 and 2014. We investigated the effects
of neutral and immune gene heterozygosity on early mortality due to
bacterial infection using the \texttt{inbreedR} package.\footnote{Stoffel,
  M. A., Esser, M., Kardos, M., Humble, E., Nichols, H., David, P., \&
  Hoffman, J. I. (2016). inbreedR: an R package for the analysis of
  inbreeding based on genetic markers. Methods in Ecology and Evolution,
  7(11), 1331-1339.}

\begin{center}\rule{0.5\linewidth}{\linethickness}\end{center}

\subsubsection{Download packages and
libraries}\label{download-packages-and-libraries}

In order to repeat analyses presented in this manuscript a number of
packages that extend the functionalities of base \texttt{R} are
required. These can be installed using the code shown below.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{install.packages}\NormalTok{(}\StringTok{'inbreedR'}\NormalTok{)}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{"Rcpp"}\NormalTok{)}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{"readxl"}\NormalTok{)}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{"ggplot2"}\NormalTok{)}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{"gridExtra"}\NormalTok{)}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{"stringi"}\NormalTok{, }\DataTypeTok{repos=}\StringTok{"http://cran.rstudio.com/"}\NormalTok{, }\DataTypeTok{dependencies=}\OtherTok{TRUE}\NormalTok{)}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{"fansi"}\NormalTok{)}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{"adegenet"}\NormalTok{)}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{"AICcmodavg"}\NormalTok{)}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{"raster"}\NormalTok{)}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{"reshape2"}\NormalTok{)}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{"kableExtra"}\NormalTok{)}
\KeywordTok{source}\NormalTok{(}\StringTok{"https://bioconductor.org/biocLite.R"}\NormalTok{)}
\KeywordTok{biocLite}\NormalTok{(}\StringTok{"qvalue"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(inbreedR)}
\KeywordTok{library}\NormalTok{(readxl)}
\KeywordTok{library}\NormalTok{(magrittr)}
\KeywordTok{library}\NormalTok{(ggplot2)}
\KeywordTok{library}\NormalTok{(grid)}
\KeywordTok{library}\NormalTok{(gridExtra)}
\KeywordTok{library}\NormalTok{(AICcmodavg)}
\KeywordTok{library}\NormalTok{(Matrix)}
\KeywordTok{library}\NormalTok{(lme4)}
\KeywordTok{library}\NormalTok{(qvalue)}
\KeywordTok{library}\NormalTok{(adegenet)}
\KeywordTok{library}\NormalTok{(reshape2)}
\KeywordTok{library}\NormalTok{(kableExtra)}
\end{Highlighting}
\end{Shaded}

In order to use \texttt{inbreedR}, the working format is typically an
\emph{individual x loci} matrix, where rows represent individuals and
every two columns represent a single locus. If an individual is
heterozygous at a given locus, it is coded as 1, whereas a homozygote is
coded as 0, and missing data are coded as NA.

The first step is to read the data from an excel file. Our original
table includes, plate number, well number, species, id, year, health
status (represented by a binomial with 0 for healthy and 1 for
infected), birth weight, and the following markers (a and b for
alleles).

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{## reda data}
\NormalTok{seals <-}\StringTok{ }\NormalTok{readxl}\OperatorTok{::}\KeywordTok{read_excel}\NormalTok{(}\StringTok{"data/genotypes_raw.xlsx"}\NormalTok{, }\DataTypeTok{skip =} \DecValTok{1}\NormalTok{)[}\DecValTok{1}\OperatorTok{:}\DecValTok{78}\NormalTok{,]}
\NormalTok{## express alleles as numerals}
\NormalTok{seals[}\DecValTok{8}\OperatorTok{:}\KeywordTok{ncol}\NormalTok{(seals)] <-}\StringTok{ }\KeywordTok{lapply}\NormalTok{(seals[}\DecValTok{8}\OperatorTok{:}\KeywordTok{ncol}\NormalTok{(seals)], as.numeric)}
\end{Highlighting}
\end{Shaded}

Here is an example of what the data frame looks like:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{head}\NormalTok{(seals[}\DecValTok{1}\OperatorTok{:}\DecValTok{6}\NormalTok{,}\DecValTok{4}\OperatorTok{:}\DecValTok{12}\NormalTok{])}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 6 x 9
##   ID    Year  `Health status` Birthweight Agt47.a Agt47.b Agt10.a Agt10.b
##   <chr> <chr> <chr>           <chr>         <dbl>   <dbl>   <dbl>   <dbl>
## 1 AGP0~ 2000  0               5.09999999~     237     245     213     213
## 2 AGP0~ 2000  1               4.8             241     245     213     213
## 3 AGP0~ 2001  1               4.8             241     241     213     213
## 4 AGP0~ 2001  0               4.45            237     241     213     215
## 5 AGP0~ 2002  0               4.59999999~     237     241     213     213
## 6 AGP0~ 2002  1               4.05            245     245     213     213
## # ... with 1 more variable: Agi11.a <dbl>
\end{verbatim}

Since demographic data is present in the beginning of the data frame, we
will start our new genotype file from the 8th column onwards. The
function \texttt{convert\_raw} converts a common format for genetic
markers (two columns per locus) into the \texttt{inbreedR} working
format. Afterwards, \texttt{check\_data} allowes to test whether the
genotype data frame has the correct format for subsequent analyses using
\texttt{inbreedR} functions.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{seals_geno <-}\StringTok{ }\KeywordTok{convert_raw}\NormalTok{(seals[}\DecValTok{8}\OperatorTok{:}\KeywordTok{ncol}\NormalTok{(seals)])}
\KeywordTok{check_data}\NormalTok{(seals_geno, }\DataTypeTok{num_ind =} \DecValTok{78}\NormalTok{, }\DataTypeTok{num_loci =} \DecValTok{61}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{center}\rule{0.5\linewidth}{\linethickness}\end{center}

\subsubsection{Separate loci by marker
type}\label{separate-loci-by-marker-type}

Divide the neutral and immune markers from their respective columns in
the adjusted inbreedR format, and compute standard multilocus
heterozygosity (sMLH).\footnote{Coltman, D. W. and J. Slate. 2003.
  Microsatellite measures of inbreeding: a meta-analysis. Evolution
  57:971--983.}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{immune_markers <-}\StringTok{ }\NormalTok{seals_geno[, }\DecValTok{1}\OperatorTok{:}\DecValTok{13}\NormalTok{]}
\NormalTok{neutral_markers <-}\StringTok{ }\NormalTok{seals_geno[, }\DecValTok{14}\OperatorTok{:}\DecValTok{61}\NormalTok{]}

\NormalTok{all_het <-}\StringTok{ }\KeywordTok{sMLH}\NormalTok{(seals_geno)}
\NormalTok{neutral_het <-}\StringTok{ }\KeywordTok{sMLH}\NormalTok{(neutral_markers)}
\NormalTok{immune_het <-}\StringTok{ }\KeywordTok{sMLH}\NormalTok{(immune_markers)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Create and reshape
dataframe}\label{create-and-reshape-dataframe}

Take out id, health, marker types, and birth weight as variables.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{birthweight <-}\StringTok{ }\KeywordTok{as.numeric}\NormalTok{(}\KeywordTok{as.character}\NormalTok{(seals[[}\StringTok{"Birthweight"}\NormalTok{]]))}

\NormalTok{sealdata <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{id =}\NormalTok{ seals[[}\DecValTok{4}\NormalTok{]],  }\DataTypeTok{health =} \KeywordTok{factor}\NormalTok{(seals[[}\DecValTok{6}\NormalTok{]]), }
                       \DataTypeTok{All =}\NormalTok{ all_het, }\DataTypeTok{Neutral =}\NormalTok{ neutral_het, }\DataTypeTok{Immune =}\NormalTok{ immune_het)}

\NormalTok{sealdataweight <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{id =}\NormalTok{ seals[[}\DecValTok{4}\NormalTok{]],  }\DataTypeTok{health =} \KeywordTok{factor}\NormalTok{(seals[[}\DecValTok{6}\NormalTok{]]), birthweight, }
                             \DataTypeTok{All =}\NormalTok{ all_het, }\DataTypeTok{Neutral =}\NormalTok{ neutral_het, }\DataTypeTok{Immune =}\NormalTok{ immune_het)}

\NormalTok{sealdata_reshaped <-}\StringTok{ }\NormalTok{reshape2}\OperatorTok{::}\KeywordTok{melt}\NormalTok{(sealdata)}
\NormalTok{sealdataframe_plusyear <-}\StringTok{ }\KeywordTok{cbind}\NormalTok{(sealdataweight, }\DataTypeTok{year =} \KeywordTok{as.numeric}\NormalTok{(seals[[}\DecValTok{5}\NormalTok{]])) }
\end{Highlighting}
\end{Shaded}

\begin{center}\rule{0.5\linewidth}{\linethickness}\end{center}

\subsubsection{\texorpdfstring{Calculate
\emph{g}\textsubscript{2}}{Calculate g2}}\label{calculate-g2}

\emph{g}\textsubscript{2} is a proxy for identity disequilibrium. It is
a measure of two-locus disequilibrium, which quantifies the extent to
which heterozygosities are correlated across pairs of loci.\footnote{David,
  P., Pujol, B., Viard, F., Castella, V., \& Goudet, J. (2007). Reliable
  selfing rate estimates from imperfect population genetic data.
  Molecular ecology, 16(12), 2474-2487.} This allows us to take a look
at our neutral marker heterozygosity to determine if there is variation
in inbreeding in the population.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{g2_neutral <-}\StringTok{ }\KeywordTok{g2_microsats}\NormalTok{(neutral_markers, }\DataTypeTok{nperm =} \DecValTok{9999}\NormalTok{, }\DataTypeTok{nboot =} \DecValTok{9999}\NormalTok{) }
\NormalTok{g2_neutral_bs <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{bs =}\NormalTok{ g2_neutral}\OperatorTok{$}\NormalTok{g2_boot,}
                            \DataTypeTok{lcl =}\NormalTok{ g2_neutral}\OperatorTok{$}\NormalTok{CI_boot[[}\DecValTok{1}\NormalTok{]],}
                            \DataTypeTok{ucl =}\NormalTok{ g2_neutral}\OperatorTok{$}\NormalTok{CI_boot[[}\DecValTok{2}\NormalTok{]],}
                            \DataTypeTok{g2  =}\NormalTok{ g2_neutral}\OperatorTok{$}\NormalTok{g2,}
                            \DataTypeTok{p =}\NormalTok{ g2_neutral}\OperatorTok{$}\NormalTok{p_val)}
\end{Highlighting}
\end{Shaded}

Plot the distribution of g2 estimates:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{g2_neutral_bs_histogram <-}
\StringTok{  }\NormalTok{ggplot2}\OperatorTok{::}\KeywordTok{ggplot}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme_classic}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_histogram}\NormalTok{(}\DataTypeTok{binwidth =} \FloatTok{0.000375}\NormalTok{, }\DataTypeTok{data =}\NormalTok{ g2_neutral_bs, }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ bs),}
                 \DataTypeTok{color =} \StringTok{"#0294A5"}\NormalTok{,}
                 \DataTypeTok{fill =} \StringTok{"#0294A5"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_errorbarh}\NormalTok{(}\DataTypeTok{data =}\NormalTok{ g2_neutral_bs,}
                 \KeywordTok{aes}\NormalTok{(}\DataTypeTok{y =} \DecValTok{1040}\NormalTok{, }\DataTypeTok{x =}\NormalTok{ g2, }\DataTypeTok{xmin =}\NormalTok{ lcl, }\DataTypeTok{xmax =}\NormalTok{ ucl),}
                 \DataTypeTok{color =} \StringTok{"black"}\NormalTok{, }\DataTypeTok{size =} \FloatTok{0.7}\NormalTok{, }\DataTypeTok{linetype =} \StringTok{"solid"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_linerange}\NormalTok{(}\DataTypeTok{data =}\NormalTok{ g2_neutral_bs,}
                 \KeywordTok{aes}\NormalTok{(}\DataTypeTok{ymin =} \DecValTok{0}\NormalTok{, }\DataTypeTok{ymax =} \DecValTok{1040}\NormalTok{, }\DataTypeTok{x =}\NormalTok{ g2),}
                 \DataTypeTok{linetype =} \StringTok{'dotted'}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme}\NormalTok{(}\DataTypeTok{text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{size =} \DecValTok{12}\NormalTok{),}
        \DataTypeTok{panel.border =} \KeywordTok{element_blank}\NormalTok{(),}
        \DataTypeTok{strip.background =}\KeywordTok{element_rect}\NormalTok{(}\DataTypeTok{fill =} \StringTok{"white"}\NormalTok{, }\DataTypeTok{colour =} \StringTok{"white"}\NormalTok{),}
        \DataTypeTok{strip.text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{colour =} \StringTok{'white'}\NormalTok{),}
        \DataTypeTok{plot.margin =}\NormalTok{ grid}\OperatorTok{::}\KeywordTok{unit}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{), }\StringTok{'mm'}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{facet_wrap}\NormalTok{(}\OperatorTok{~}\NormalTok{p) }\OperatorTok{+}
\StringTok{  }\KeywordTok{ylab}\NormalTok{(}\StringTok{"Counts"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{labs}\NormalTok{(}\DataTypeTok{x =} \KeywordTok{expression}\NormalTok{(}\KeywordTok{italic}\NormalTok{(g)[}\StringTok{"2"}\NormalTok{])) }\OperatorTok{+}
\StringTok{  }\KeywordTok{ggtitle}\NormalTok{(}\StringTok{"a)"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_y_continuous}\NormalTok{(}\DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{), }\DataTypeTok{limits =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{1200}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_x_continuous}\NormalTok{(}\DataTypeTok{limits =} \KeywordTok{c}\NormalTok{(}\OperatorTok{-}\FloatTok{0.003}\NormalTok{, }\FloatTok{0.007}\NormalTok{),}
                     \DataTypeTok{breaks =} \KeywordTok{seq}\NormalTok{(}\OperatorTok{-}\FloatTok{0.003}\NormalTok{, }\FloatTok{0.009}\NormalTok{, }\FloatTok{0.003}\NormalTok{),}
                     \DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{annotate}\NormalTok{(}\StringTok{"text"}\NormalTok{, }\DataTypeTok{x =}\NormalTok{ g2_neutral_bs}\OperatorTok{$}\NormalTok{g2, }\DataTypeTok{y =} \DecValTok{1100}\NormalTok{,}
           \DataTypeTok{label =} \KeywordTok{paste0}\NormalTok{(}\StringTok{'p = '}\NormalTok{, }\KeywordTok{round}\NormalTok{(g2_neutral_bs}\OperatorTok{$}\NormalTok{p, }\DecValTok{3}\NormalTok{)),}
           \DataTypeTok{family =} \KeywordTok{theme_get}\NormalTok{()}\OperatorTok{$}\NormalTok{text[[}\StringTok{"family"}\NormalTok{]],}
           \DataTypeTok{size =} \KeywordTok{theme_get}\NormalTok{()}\OperatorTok{$}\NormalTok{text[[}\StringTok{"size"}\NormalTok{]]}\OperatorTok{/}\DecValTok{4}\NormalTok{) }
\KeywordTok{plot}\NormalTok{(g2_neutral_bs_histogram)}
\end{Highlighting}
\end{Shaded}

\includegraphics{R-code_files/figure-latex/neutral g2 plot-1.pdf}

\begin{center}\rule{0.5\linewidth}{\linethickness}\end{center}

\subsubsection{Plot heterozygosity among marker
sets}\label{plot-heterozygosity-among-marker-sets}

In order to visualize sMLH for all, neutral, and immune markers, create
the following box-plot:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{het_plot <-}\StringTok{ }
\StringTok{  }\KeywordTok{ggplot}\NormalTok{(}\DataTypeTok{data =}\NormalTok{ sealdata_reshaped, }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ health, }\DataTypeTok{y =}\NormalTok{ value, }\DataTypeTok{fill =}\NormalTok{ variable)) }\OperatorTok{+}\StringTok{ }
\StringTok{  }\KeywordTok{stat_boxplot}\NormalTok{(}\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ health, }\DataTypeTok{y =}\NormalTok{ value), }
               \DataTypeTok{geom =} \StringTok{'errorbar'}\NormalTok{, }\DataTypeTok{linetype =} \DecValTok{1}\NormalTok{, }\DataTypeTok{width =} \FloatTok{0.5}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_boxplot}\NormalTok{( }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ health, }\DataTypeTok{y =}\NormalTok{ value), }\DataTypeTok{outlier.shape =} \DecValTok{1}\NormalTok{) }\OperatorTok{+}\StringTok{    }
\StringTok{  }\KeywordTok{geom_jitter}\NormalTok{(}\DataTypeTok{shape =} \DecValTok{16}\NormalTok{, }\DataTypeTok{position =} \KeywordTok{position_jitter}\NormalTok{(}\FloatTok{0.2}\NormalTok{), }\DataTypeTok{size =}\NormalTok{ .}\DecValTok{8}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme_classic}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme}\NormalTok{(}\DataTypeTok{legend.position =} \StringTok{"none"}\NormalTok{,}
        \DataTypeTok{panel.border =} \KeywordTok{element_blank}\NormalTok{(),}
        \DataTypeTok{strip.background =} \KeywordTok{element_blank}\NormalTok{(),}
        \DataTypeTok{text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{size =} \DecValTok{12}\NormalTok{),}
        \DataTypeTok{plot.margin =}\NormalTok{ grid}\OperatorTok{::}\KeywordTok{unit}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{), }\StringTok{'mm'}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{xlab}\NormalTok{(}\StringTok{"Infection status"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{ylab}\NormalTok{(}\StringTok{"sMLH"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{ggtitle}\NormalTok{(}\StringTok{"b)"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_fill_manual}\NormalTok{(}\DataTypeTok{values =}  \KeywordTok{c}\NormalTok{(}\StringTok{"#A79C93"}\NormalTok{, }\StringTok{"#0294A5"}\NormalTok{, }\StringTok{"#C1403D"}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{facet_wrap}\NormalTok{(}\OperatorTok{~}\NormalTok{variable, }\DataTypeTok{nrow =} \DecValTok{1}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_y_continuous}\NormalTok{(}\DataTypeTok{limits =} \KeywordTok{c}\NormalTok{(}\FloatTok{0.3}\NormalTok{, }\DecValTok{2}\NormalTok{),}
                     \DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{))}
\KeywordTok{plot}\NormalTok{(het_plot)}
\end{Highlighting}
\end{Shaded}

\includegraphics{R-code_files/figure-latex/unnamed-chunk-5-1.pdf}

\begin{center}\rule{0.5\linewidth}{\linethickness}\end{center}

\subsubsection{Calculate heterozygosity for each individual
locus}\label{calculate-heterozygosity-for-each-individual-locus}

As we have previously looked at genome-wide effects, it may be of
interest to look for local effects. Therefore, we wanted to examine the
heterozygosity for each locus. First, define a function tomput the
confidence interval:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{confidence_interval <-}\StringTok{ }\ControlFlowTok{function}\NormalTok{(vector) \{}
\NormalTok{  vec_sd <-}\StringTok{ }\KeywordTok{sd}\NormalTok{(vector)        }\CommentTok{# standard deviation of sample}
\NormalTok{  n <-}\StringTok{ }\KeywordTok{length}\NormalTok{(vector)         }\CommentTok{# sample size}
\NormalTok{  vec_mean <-}\StringTok{ }\KeywordTok{mean}\NormalTok{(vector)    }\CommentTok{# mean of sample}
\NormalTok{  error <-}\StringTok{ }\KeywordTok{qt}\NormalTok{((.}\DecValTok{95} \OperatorTok{+}\StringTok{ }\DecValTok{1}\NormalTok{)}\OperatorTok{/}\DecValTok{2}\NormalTok{, }\DataTypeTok{df =}\NormalTok{ n }\OperatorTok{-}\StringTok{ }\DecValTok{1}\NormalTok{) }\OperatorTok{*}\StringTok{ }\NormalTok{vec_sd }\OperatorTok{/}\StringTok{ }\KeywordTok{sqrt}\NormalTok{(n)               }\CommentTok{# error according to t distribution}
\NormalTok{  result <-}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"lower"}\NormalTok{ =}\StringTok{ }\NormalTok{vec_mean }\OperatorTok{-}\StringTok{ }\NormalTok{error, }\StringTok{"upper"}\NormalTok{ =}\StringTok{ }\NormalTok{vec_mean }\OperatorTok{+}\StringTok{ }\NormalTok{error)   }\CommentTok{# confidence interval as a vector}
  \KeywordTok{return}\NormalTok{(result)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

Calculate the heterozygosity for each locus, and use a regression on
infection status:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{## calcaute sMLH}
\NormalTok{het_per_locus <-}\StringTok{ }\KeywordTok{apply}\NormalTok{(seals_geno, }\DecValTok{2}\NormalTok{, sMLH)}
\NormalTok{## add factors }
\NormalTok{df <-}\StringTok{  }\KeywordTok{cbind}\NormalTok{(sealdataframe_plusyear, seals_geno) }
\NormalTok{## add marker type as names to the data.frame}
\KeywordTok{names}\NormalTok{(df)[}\DecValTok{6}\OperatorTok{:}\DecValTok{66}\NormalTok{] <-}\StringTok{ }\KeywordTok{c}\NormalTok{(}\KeywordTok{paste0}\NormalTok{(}\StringTok{"Immune"}\NormalTok{, }\DecValTok{1}\OperatorTok{:}\DecValTok{13}\NormalTok{), }\KeywordTok{paste0}\NormalTok{(}\StringTok{"Neutral"}\NormalTok{, }\DecValTok{1}\OperatorTok{:}\DecValTok{48}\NormalTok{))}

\NormalTok{lm_by_loc <-}\StringTok{ }\KeywordTok{lapply}\NormalTok{(}\DecValTok{1}\OperatorTok{:}\DecValTok{61}\NormalTok{, }\ControlFlowTok{function}\NormalTok{(x) \{}
\NormalTok{  value <-}\StringTok{ }\NormalTok{df[,x }\OperatorTok{+}\StringTok{ }\DecValTok{7}\NormalTok{]                                 }\CommentTok{# extract data of given marker x}
  \CommentTok{# res <- summary(lme4::lmer(as.numeric(df$health) ~ value + (1|df$year)))   # do regression}
  \CommentTok{# conf <- confint(lme4::lmer(as.numeric(df$health) ~ value + (1|df$year)))}
\NormalTok{  res <-}\StringTok{ }\KeywordTok{summary}\NormalTok{(}\KeywordTok{lm}\NormalTok{(}\KeywordTok{as.numeric}\NormalTok{(df}\OperatorTok{$}\NormalTok{health) }\OperatorTok{~}\StringTok{ }\NormalTok{value))   }\CommentTok{# do regression}
\NormalTok{  conf <-}\StringTok{ }\KeywordTok{confint}\NormalTok{(}\KeywordTok{lm}\NormalTok{(}\KeywordTok{as.numeric}\NormalTok{(df}\OperatorTok{$}\NormalTok{health) }\OperatorTok{~}\StringTok{ }\NormalTok{value))}
\NormalTok{  f <-}\StringTok{ }\NormalTok{res}\OperatorTok{$}\NormalTok{fstatistic}
  \KeywordTok{pf}\NormalTok{(f[}\DecValTok{1}\NormalTok{], f[}\DecValTok{2}\NormalTok{], f[}\DecValTok{3}\NormalTok{], }\DataTypeTok{lower=}\OtherTok{FALSE}\NormalTok{)}
\NormalTok{  out <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{beta =}\NormalTok{ res}\OperatorTok{$}\NormalTok{coefficients[}\DecValTok{2}\NormalTok{,}\DecValTok{1}\NormalTok{],}
                    \DataTypeTok{lcl =}\NormalTok{ conf[}\DecValTok{2}\NormalTok{,}\DecValTok{1}\NormalTok{],}
                    \DataTypeTok{ucl =}\NormalTok{ conf[}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{])}
\NormalTok{\}) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{do.call}\NormalTok{(}\StringTok{"rbind"}\NormalTok{,.) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{cbind}\NormalTok{(., }\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{names =} \KeywordTok{colnames}\NormalTok{(seals)[}\KeywordTok{seq}\NormalTok{(}\DecValTok{8}\NormalTok{, }\KeywordTok{ncol}\NormalTok{(seals), }\DecValTok{2}\NormalTok{)] }\OperatorTok{%>%}\StringTok{ }
\StringTok{                          }\KeywordTok{substring}\NormalTok{(., }\DataTypeTok{first =} \DecValTok{1}\NormalTok{, }\DataTypeTok{last =} \KeywordTok{nchar}\NormalTok{(.) }\OperatorTok{-}\StringTok{ }\DecValTok{2}\NormalTok{),}
                      \DataTypeTok{type =} \KeywordTok{c}\NormalTok{(}\KeywordTok{rep}\NormalTok{(}\StringTok{"Immune"}\NormalTok{, }\DecValTok{13}\NormalTok{),}\KeywordTok{rep}\NormalTok{(}\StringTok{"Neutral"}\NormalTok{, }\DecValTok{48}\NormalTok{)),}
                      \DataTypeTok{dummy =} \StringTok{""}\NormalTok{))}

 \CommentTok{# order by effect size}
\NormalTok{lm_by_loc <-}\StringTok{ }\NormalTok{lm_by_loc[}\KeywordTok{with}\NormalTok{(lm_by_loc, }\KeywordTok{order}\NormalTok{(type, beta, }\DataTypeTok{decreasing =}\NormalTok{ F)),]   }
\NormalTok{lm_by_loc}\OperatorTok{$}\NormalTok{num <-}\StringTok{ }\DecValTok{1}\OperatorTok{:}\DecValTok{61}

\NormalTok{## create data frame to label effects}
\NormalTok{names_df <-}\StringTok{  }\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{label =}\NormalTok{ lm_by_loc}\OperatorTok{$}\NormalTok{names,}
                        \DataTypeTok{num =}\NormalTok{ lm_by_loc}\OperatorTok{$}\NormalTok{num)}
\end{Highlighting}
\end{Shaded}

Create a plot to feature each loci and their relevant effect sizes:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{het_by_loci_plot <-}\StringTok{ }\KeywordTok{ggplot}\NormalTok{(lm_by_loc, }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ num, }\DataTypeTok{y =}\NormalTok{ beta, }\DataTypeTok{col =}\NormalTok{ type)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_errorbar}\NormalTok{(}\KeywordTok{aes}\NormalTok{(}\DataTypeTok{ymin =}\NormalTok{ lcl, }\DataTypeTok{ymax =}\NormalTok{ ucl),}
                \DataTypeTok{width =} \FloatTok{0.6}\NormalTok{, }\DataTypeTok{alpha =} \FloatTok{0.7}\NormalTok{, }\DataTypeTok{size =} \FloatTok{0.7}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_point}\NormalTok{(}\DataTypeTok{size =} \DecValTok{1}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_x_continuous}\NormalTok{(}\DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{), }\DataTypeTok{breaks =} \DecValTok{1}\OperatorTok{:}\DecValTok{61}\NormalTok{, }\DataTypeTok{labels =}\NormalTok{ names_df}\OperatorTok{$}\NormalTok{label) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_y_continuous}\NormalTok{(}\DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_hline}\NormalTok{(}\DataTypeTok{yintercept =} \DecValTok{0}\NormalTok{, }\DataTypeTok{linetype =} \StringTok{"dotted"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{coord_flip}\NormalTok{(}\DataTypeTok{xlim =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{, }\FloatTok{61.5}\NormalTok{), }\DataTypeTok{ylim =} \KeywordTok{c}\NormalTok{(}\OperatorTok{-}\DecValTok{1}\NormalTok{,}\DecValTok{1}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_color_manual}\NormalTok{(}\DataTypeTok{values =} \KeywordTok{c}\NormalTok{(}\StringTok{"#C1403D"}\NormalTok{,}\StringTok{"#0294A5"}\NormalTok{), }
                     \DataTypeTok{name =} \StringTok{""}\NormalTok{,}
                     \DataTypeTok{breaks =} \KeywordTok{c}\NormalTok{(}\StringTok{"Neutral"}\NormalTok{, }\StringTok{"Immune"}\NormalTok{),}
                     \DataTypeTok{labels =} \KeywordTok{c}\NormalTok{(}\StringTok{"Neutral"}\NormalTok{, }\StringTok{"Immune"}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme_classic}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{xlab}\NormalTok{(}\StringTok{""}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{ylab}\NormalTok{(}\StringTok{"Effect size"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme}\NormalTok{(}\DataTypeTok{legend.justification =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{1}\NormalTok{),}
        \DataTypeTok{legend.position =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\FloatTok{1.05}\NormalTok{),}
        \DataTypeTok{legend.background =} \KeywordTok{element_rect}\NormalTok{(}\DataTypeTok{fill =} \OtherTok{NA}\NormalTok{),}
        \DataTypeTok{text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{size =} \DecValTok{12}\NormalTok{),}
        \DataTypeTok{axis.text.y =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{size =} \DecValTok{5}\NormalTok{),}
        \DataTypeTok{legend.text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{size =} \DecValTok{7}\NormalTok{),}
        \DataTypeTok{panel.border =} \KeywordTok{element_blank}\NormalTok{(),}
        \DataTypeTok{strip.background =} \KeywordTok{element_rect}\NormalTok{(}\DataTypeTok{fill =} \StringTok{"white"}\NormalTok{, }\DataTypeTok{colour =} \StringTok{"white"}\NormalTok{),}
        \DataTypeTok{strip.text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{colour =} \StringTok{'white'}\NormalTok{),}
        \DataTypeTok{plot.margin =}\NormalTok{ grid}\OperatorTok{::}\KeywordTok{unit}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{), }\StringTok{'mm'}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{guides}\NormalTok{(}\DataTypeTok{color =} \KeywordTok{guide_legend}\NormalTok{(}
    \DataTypeTok{keywidth =} \FloatTok{0.05}\NormalTok{,}
    \DataTypeTok{keyheight =} \FloatTok{0.05}\NormalTok{,}
    \DataTypeTok{default.unit =} \StringTok{"inch"}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{facet_wrap}\NormalTok{(}\OperatorTok{~}\NormalTok{dummy) }\OperatorTok{+}
\StringTok{  }\KeywordTok{ggtitle}\NormalTok{(}\StringTok{"c)"}\NormalTok{)}
\NormalTok{het_by_loci_plot}
\end{Highlighting}
\end{Shaded}

\includegraphics{R-code_files/figure-latex/unnamed-chunk-7-1.pdf}

To look for local effects between effect sizes of the neutral and immune
loci, use a Wilcoxon test:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{wilcox.test}\NormalTok{(lm_by_loc}\OperatorTok{$}\NormalTok{beta[}\DecValTok{1}\OperatorTok{:}\DecValTok{13}\NormalTok{],lm_by_loc}\OperatorTok{$}\NormalTok{beta[}\DecValTok{14}\OperatorTok{:}\DecValTok{61}\NormalTok{])}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Wilcoxon rank sum test
## 
## data:  lm_by_loc$beta[1:13] and lm_by_loc$beta[14:61]
## W = 285, p-value = 0.6445
## alternative hypothesis: true location shift is not equal to 0
\end{verbatim}

\begin{center}\rule{0.5\linewidth}{\linethickness}\end{center}

\subsubsection{Create the final combined
figure}\label{create-the-final-combined-figure}

To create a combination plot of all figures (as in the manuscript):

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{lay <-}\StringTok{ }\KeywordTok{rbind}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{,}\DecValTok{3}\NormalTok{),}
             \KeywordTok{c}\NormalTok{(}\DecValTok{2}\NormalTok{,}\DecValTok{3}\NormalTok{))}

\NormalTok{combo_plot <-}\StringTok{ }\KeywordTok{grid.arrange}\NormalTok{(g2_neutral_bs_histogram, }
\NormalTok{                           het_plot, }
\NormalTok{                           het_by_loci_plot, }\DataTypeTok{ncol =} \DecValTok{3}\NormalTok{, }\DataTypeTok{layout_matrix =}\NormalTok{ lay)}
\end{Highlighting}
\end{Shaded}

\includegraphics{R-code_files/figure-latex/create combo_plot-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{combo_plot}
\end{Highlighting}
\end{Shaded}

\begin{center}\rule{0.5\linewidth}{\linethickness}\end{center}

\subsubsection{Modeling}\label{modeling}

To test for associations between microsatellite heterozygosity and death
from bacterial infection, we constructed several alternative generalized
linear mixed-models (GLMMs) incorporating relevant predictor variables
and quantified their relative support using AICc weights within a
multi-model inference framework. All of the models had pup survival as a
binary response variable (coded as 0 = alive and 1 = dead) and included
year as a random effect to statistically control for any variation in
survivorship attributable to inter-annual variation. The following GLMMs
were considered:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{models <-}\StringTok{ }\KeywordTok{list}\NormalTok{(}
  \KeywordTok{glmer}\NormalTok{(health }\OperatorTok{~}\StringTok{  }\DecValTok{1} \OperatorTok{+}\StringTok{ }\NormalTok{(}\DecValTok{1}\OperatorTok{|}\NormalTok{year), }\DataTypeTok{data =}\NormalTok{ sealdataframe_plusyear, }\DataTypeTok{family =} \StringTok{'binomial'}\NormalTok{),}
  \KeywordTok{glmer}\NormalTok{(health }\OperatorTok{~}\StringTok{  }\NormalTok{All }\OperatorTok{+}\StringTok{ }\NormalTok{(}\DecValTok{1}\OperatorTok{|}\NormalTok{year), }\DataTypeTok{data =}\NormalTok{ sealdataframe_plusyear, }\DataTypeTok{family =} \StringTok{'binomial'}\NormalTok{),}
  \KeywordTok{glmer}\NormalTok{(health }\OperatorTok{~}\StringTok{  }\NormalTok{Immune }\OperatorTok{+}\StringTok{ }\NormalTok{(}\DecValTok{1}\OperatorTok{|}\NormalTok{year), }\DataTypeTok{data =}\NormalTok{ sealdataframe_plusyear, }\DataTypeTok{family =} \StringTok{'binomial'}\NormalTok{),}
  \KeywordTok{glmer}\NormalTok{(health }\OperatorTok{~}\StringTok{  }\NormalTok{Neutral }\OperatorTok{+}\StringTok{ }\NormalTok{(}\DecValTok{1}\OperatorTok{|}\NormalTok{year), }\DataTypeTok{data =}\NormalTok{ sealdataframe_plusyear, }\DataTypeTok{family =} \StringTok{'binomial'}\NormalTok{), }
  \KeywordTok{glmer}\NormalTok{(health }\OperatorTok{~}\StringTok{  }\DecValTok{1} \OperatorTok{+}\StringTok{ }\NormalTok{birthweight }\OperatorTok{+}\StringTok{ }\NormalTok{(}\DecValTok{1}\OperatorTok{|}\NormalTok{year), }\DataTypeTok{data =}\NormalTok{ sealdataframe_plusyear, }\DataTypeTok{family =} \StringTok{'binomial'}\NormalTok{),}
  \KeywordTok{glmer}\NormalTok{(health }\OperatorTok{~}\StringTok{  }\NormalTok{All }\OperatorTok{+}\StringTok{ }\NormalTok{birthweight }\OperatorTok{+}\StringTok{ }\NormalTok{(}\DecValTok{1}\OperatorTok{|}\NormalTok{year), }\DataTypeTok{data =}\NormalTok{ sealdataframe_plusyear, }\DataTypeTok{family =} \StringTok{'binomial'}\NormalTok{),}
  \KeywordTok{glmer}\NormalTok{(health }\OperatorTok{~}\StringTok{  }\NormalTok{Immune }\OperatorTok{+}\StringTok{ }\NormalTok{birthweight }\OperatorTok{+}\StringTok{ }\NormalTok{(}\DecValTok{1}\OperatorTok{|}\NormalTok{year), }\DataTypeTok{data =}\NormalTok{ sealdataframe_plusyear, }\DataTypeTok{family =} \StringTok{'binomial'}\NormalTok{),}
  \KeywordTok{glmer}\NormalTok{(health }\OperatorTok{~}\StringTok{  }\NormalTok{Neutral }\OperatorTok{+}\StringTok{ }\NormalTok{birthweight }\OperatorTok{+}\StringTok{ }\NormalTok{(}\DecValTok{1}\OperatorTok{|}\NormalTok{year), }\DataTypeTok{data =}\NormalTok{ sealdataframe_plusyear, }\DataTypeTok{family =} \StringTok{'binomial'}\NormalTok{)) }
\KeywordTok{names}\NormalTok{(models) <-}\StringTok{ }\KeywordTok{paste0}\NormalTok{(}\StringTok{"m"}\NormalTok{, }\DecValTok{1}\OperatorTok{:}\KeywordTok{length}\NormalTok{(models))}

\NormalTok{## model selection}
\NormalTok{kableExtra}\OperatorTok{::}\KeywordTok{kable}\NormalTok{(AICcmodavg}\OperatorTok{::}\KeywordTok{aictab}\NormalTok{(models, }\DataTypeTok{second.ord =}\NormalTok{ T), }\DataTypeTok{booktabs =} \OtherTok{TRUE}\NormalTok{,}
                  \DataTypeTok{longtable =} \OtherTok{FALSE}\NormalTok{, }\DataTypeTok{caption =} \StringTok{"Model selection"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:AIC models}Model selection}
\centering
\begin{tabular}[t]{llrrrrrrr}
\toprule
  & Modnames & K & AICc & Delta\_AICc & ModelLik & AICcWt & LL & Cum.Wt\\
\midrule
1 & m1 & 2 & 112.2910 & 0.000000 & 1.0000000 & 0.3440666 & -54.06548 & 0.3440666\\
5 & m5 & 3 & 114.0681 & 1.777108 & 0.4112501 & 0.1414974 & -53.87187 & 0.4855641\\
3 & m3 & 3 & 114.2755 & 1.984579 & 0.3707270 & 0.1275548 & -53.97561 & 0.6131188\\
2 & m2 & 3 & 114.4032 & 2.112277 & 0.3477962 & 0.1196651 & -54.03946 & 0.7327839\\
4 & m4 & 3 & 114.4420 & 2.151058 & 0.3411173 & 0.1173671 & -54.05885 & 0.8501510\\
\addlinespace
7 & m7 & 4 & 115.9991 & 3.708187 & 0.1565949 & 0.0538791 & -53.72560 & 0.9040300\\
6 & m6 & 4 & 116.1749 & 3.883978 & 0.1434184 & 0.0493455 & -53.81350 & 0.9533755\\
8 & m8 & 4 & 116.2884 & 3.997419 & 0.1355100 & 0.0466245 & -53.87022 & 1.0000000\\
\bottomrule
\end{tabular}
\end{table}

These included `null models' without any genetic effects (models i and
v) as well as models that included sMLH combined over all loci or
calculated separately for the neutral versus immune loci. Models v to
viii also included pup birth weight (in kg) to incorporate any potential
effects of body size on survivorship. All of the models were specified
using the glmer function of the package ``lme4'' with a binomial error
structure.\footnote{Bates, D., Mächler, M., Bolker, B., \& Walker, S.
  (2014). Fitting linear mixed-effects models using lme4. arXiv preprint
  arXiv:1406.5823.} Using the R package \texttt{AICcmodavg}, the most
parsimonious model was selected based on the delta AICc value, which
compares weights as a measure of the likelihood of a particular
model.\footnote{Mazerolle, M. J., \& Mazerolle, M. M. J. (2017). Package
  `AICcmodavg'. R package.} The best supported model has \(\Delta\) AICc
= 0 and a difference of two or more units was applied as a criterion for
choosing one model over a competing model.\footnote{Anderson, D. R., \&
  Burnham, K. P. (2002). Avoiding pitfalls when using
  information-theoretic methods. The Journal of Wildlife Management,
  912-918.}

Apply a false discovery rate correction for a table of p-values.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pval <-}\StringTok{ }\KeywordTok{read.table}\NormalTok{(}\StringTok{"pvalues.txt"}\NormalTok{,}\DataTypeTok{header =}\NormalTok{ F, }\DataTypeTok{sep =} \StringTok{","}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }\KeywordTok{as.vector}\NormalTok{() }\OperatorTok{%>%}\StringTok{ }\NormalTok{.[[}\DecValTok{1}\NormalTok{]]}
\NormalTok{qobj <-}\StringTok{ }\KeywordTok{qvalue}\NormalTok{(pval)}

\NormalTok{qvalues <-}\StringTok{ }\NormalTok{qobj}\OperatorTok{$}\NormalTok{qvalues}
\NormalTok{pi0 <-}\StringTok{ }\NormalTok{qobj}\OperatorTok{$}\NormalTok{pi0}
\NormalTok{lfdr <-}\StringTok{ }\NormalTok{qobj}\OperatorTok{$}\NormalTok{lfdr}
\KeywordTok{summary}\NormalTok{(qobj)}

\NormalTok{df <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{p =}\NormalTok{ qobj}\OperatorTok{$}\NormalTok{pvalues,}
                 \DataTypeTok{q =}\NormalTok{ qobj}\OperatorTok{$}\NormalTok{qvalues)}
\CommentTok{#view(df)}
\end{Highlighting}
\end{Shaded}

\begin{center}\rule{0.5\linewidth}{\linethickness}\end{center}

\section{Supplementary Data}\label{supplementary-data}

\paragraph{\texorpdfstring{(A) \emph{g}\textsubscript{2} for all marker
sets.}{(A) g2 for all marker sets.}}\label{a-g2-for-all-marker-sets.}

If there is interest to see if a variation in inbreeding can be captured
among different marker sets, calculate \emph{g}\textsubscript{2} for all
and immune microsats and create histograms:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{g2_all <-}\StringTok{ }\KeywordTok{g2_microsats}\NormalTok{(}\KeywordTok{cbind}\NormalTok{(neutral_markers, immune_markers), }\DataTypeTok{nperm =} \DecValTok{9999}\NormalTok{, }\DataTypeTok{nboot =} \DecValTok{9999}\NormalTok{)}
\NormalTok{g2_all_bs <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{bs =}\NormalTok{ g2_all}\OperatorTok{$}\NormalTok{g2_boot,}
                        \DataTypeTok{lcl =}\NormalTok{ g2_all}\OperatorTok{$}\NormalTok{CI_boot[[}\DecValTok{1}\NormalTok{]],}
                        \DataTypeTok{ucl =}\NormalTok{ g2_all}\OperatorTok{$}\NormalTok{CI_boot[[}\DecValTok{2}\NormalTok{]],}
                        \DataTypeTok{g2  =}\NormalTok{ g2_all}\OperatorTok{$}\NormalTok{g2,}
                        \DataTypeTok{p =}\NormalTok{ g2_all}\OperatorTok{$}\NormalTok{p_val)}

\NormalTok{g2_immune <-}\StringTok{ }\KeywordTok{g2_microsats}\NormalTok{(immune_markers, }\DataTypeTok{nperm =} \DecValTok{9999}\NormalTok{, }\DataTypeTok{nboot =} \DecValTok{9999}\NormalTok{)}
\NormalTok{g2_immune_bs <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{bs =}\NormalTok{ g2_immune}\OperatorTok{$}\NormalTok{g2_boot,}
                           \DataTypeTok{lcl =}\NormalTok{ g2_immune}\OperatorTok{$}\NormalTok{CI_boot[[}\DecValTok{1}\NormalTok{]],}
                           \DataTypeTok{ucl =}\NormalTok{ g2_immune}\OperatorTok{$}\NormalTok{CI_boot[[}\DecValTok{2}\NormalTok{]],}
                           \DataTypeTok{g2  =}\NormalTok{ g2_immune}\OperatorTok{$}\NormalTok{g2,}
                           \DataTypeTok{p =}\NormalTok{ g2_immune}\OperatorTok{$}\NormalTok{p_val)}

\NormalTok{all_graphs_g2_neutral_bs_histogram <-}
\StringTok{  }\NormalTok{ggplot2}\OperatorTok{::}\KeywordTok{ggplot}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme_classic}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_histogram}\NormalTok{(}\DataTypeTok{binwidth =} \FloatTok{0.000375}\NormalTok{, }\DataTypeTok{data =}\NormalTok{ g2_neutral_bs, }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ bs),}
                 \DataTypeTok{color =} \StringTok{"#0294A5"}\NormalTok{,}
                 \DataTypeTok{fill =} \StringTok{"#0294A5"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_errorbarh}\NormalTok{(}\DataTypeTok{data =}\NormalTok{ g2_neutral_bs,}
                 \KeywordTok{aes}\NormalTok{(}\DataTypeTok{y =} \DecValTok{1050}\NormalTok{, }\DataTypeTok{x =}\NormalTok{ g2, }\DataTypeTok{xmin =}\NormalTok{ lcl, }\DataTypeTok{xmax =}\NormalTok{ ucl),}
                 \DataTypeTok{color =} \StringTok{"black"}\NormalTok{, }\DataTypeTok{size =} \FloatTok{0.7}\NormalTok{, }\DataTypeTok{linetype =} \StringTok{"solid"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_linerange}\NormalTok{(}\DataTypeTok{data =}\NormalTok{ g2_neutral_bs,}
                 \KeywordTok{aes}\NormalTok{(}\DataTypeTok{ymin =} \DecValTok{0}\NormalTok{, }\DataTypeTok{ymax =} \DecValTok{1050}\NormalTok{, }\DataTypeTok{x =}\NormalTok{ g2),}
                 \DataTypeTok{linetype =} \StringTok{'dotted'}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme}\NormalTok{(}\DataTypeTok{text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{size =} \DecValTok{12}\NormalTok{),}
        \DataTypeTok{panel.border =} \KeywordTok{element_blank}\NormalTok{(),}
        \DataTypeTok{strip.background =} \KeywordTok{element_rect}\NormalTok{(}\DataTypeTok{fill =} \StringTok{"white"}\NormalTok{, }\DataTypeTok{colour =} \StringTok{"white"}\NormalTok{),}
        \DataTypeTok{strip.text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{colour =} \StringTok{'white'}\NormalTok{),}
        \DataTypeTok{plot.margin =}\NormalTok{ grid}\OperatorTok{::}\KeywordTok{unit}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{), }\StringTok{'mm'}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{facet_wrap}\NormalTok{(}\OperatorTok{~}\NormalTok{p) }\OperatorTok{+}
\StringTok{  }\KeywordTok{ylab}\NormalTok{(}\StringTok{" "}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{labs}\NormalTok{(}\DataTypeTok{x =} \KeywordTok{expression}\NormalTok{(}\KeywordTok{italic}\NormalTok{(g)[}\StringTok{"2"}\NormalTok{])) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_y_continuous}\NormalTok{(}\DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{), }\DataTypeTok{limits =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{1200}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_x_continuous}\NormalTok{(}\DataTypeTok{limits =} \KeywordTok{c}\NormalTok{(}\OperatorTok{-}\FloatTok{0.003}\NormalTok{, }\FloatTok{0.007}\NormalTok{),}
                     \DataTypeTok{breaks =} \KeywordTok{seq}\NormalTok{(}\OperatorTok{-}\FloatTok{0.003}\NormalTok{, }\FloatTok{0.009}\NormalTok{, }\FloatTok{0.003}\NormalTok{),}
                     \DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{annotate}\NormalTok{(}\StringTok{"text"}\NormalTok{, }\DataTypeTok{x =}\NormalTok{ g2_neutral_bs}\OperatorTok{$}\NormalTok{g2, }\DataTypeTok{y =} \DecValTok{1079}\NormalTok{,}
           \DataTypeTok{label =} \KeywordTok{paste0}\NormalTok{(}\StringTok{'p = '}\NormalTok{, }\KeywordTok{round}\NormalTok{(g2_neutral_bs}\OperatorTok{$}\NormalTok{p, }\DecValTok{3}\NormalTok{)),}
           \DataTypeTok{family =} \KeywordTok{theme_get}\NormalTok{()}\OperatorTok{$}\NormalTok{text[[}\StringTok{"family"}\NormalTok{]],}
           \DataTypeTok{size =} \KeywordTok{theme_get}\NormalTok{()}\OperatorTok{$}\NormalTok{text[[}\StringTok{"size"}\NormalTok{]]}\OperatorTok{/}\DecValTok{4}\NormalTok{) }

\NormalTok{all_graphs_g2_all_bs_histogram <-}
\StringTok{  }\NormalTok{ggplot2}\OperatorTok{::}\KeywordTok{ggplot}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme_classic}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_histogram}\NormalTok{(}\DataTypeTok{binwidth =} \FloatTok{0.00038}\NormalTok{, }\DataTypeTok{data =}\NormalTok{ g2_all_bs, }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ bs),}
                 \DataTypeTok{color =} \StringTok{"#A79C93"}\NormalTok{,}
                 \DataTypeTok{fill =} \StringTok{"#A79C93"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_errorbarh}\NormalTok{(}\DataTypeTok{data =}\NormalTok{ g2_all_bs,}
                 \KeywordTok{aes}\NormalTok{(}\DataTypeTok{y =} \DecValTok{1300}\NormalTok{, }\DataTypeTok{x =}\NormalTok{ g2, }\DataTypeTok{xmin =}\NormalTok{ lcl, }\DataTypeTok{xmax =}\NormalTok{ ucl),}
                 \DataTypeTok{color =} \StringTok{"black"}\NormalTok{, }\DataTypeTok{size =} \FloatTok{0.7}\NormalTok{, }\DataTypeTok{linetype =} \StringTok{"solid"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_linerange}\NormalTok{(}\DataTypeTok{data =}\NormalTok{ g2_all_bs,  }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{ymin =} \DecValTok{0}\NormalTok{, }\DataTypeTok{ymax =} \DecValTok{1300}\NormalTok{, }\DataTypeTok{x =}\NormalTok{ g2),}
                 \DataTypeTok{linetype =} \StringTok{'dotted'}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme}\NormalTok{(}\DataTypeTok{text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{size =} \DecValTok{12}\NormalTok{),}
        \DataTypeTok{panel.border =} \KeywordTok{element_blank}\NormalTok{(),}
        \DataTypeTok{strip.background =} \KeywordTok{element_rect}\NormalTok{(}\DataTypeTok{fill =} \StringTok{"white"}\NormalTok{, }\DataTypeTok{colour =} \StringTok{"white"}\NormalTok{),}
        \DataTypeTok{strip.text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{colour =} \StringTok{'white'}\NormalTok{),}
        \DataTypeTok{plot.margin =}\NormalTok{ grid}\OperatorTok{::}\KeywordTok{unit}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{), }\StringTok{'mm'}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{facet_wrap}\NormalTok{(}\OperatorTok{~}\NormalTok{p) }\OperatorTok{+}
\StringTok{  }\KeywordTok{ylab}\NormalTok{(}\StringTok{"Counts"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{xlab}\NormalTok{(}\StringTok{" "}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_y_continuous}\NormalTok{(}\DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{), }\DataTypeTok{limits =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{1500}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_x_continuous}\NormalTok{(}\DataTypeTok{limits =} \KeywordTok{c}\NormalTok{(}\OperatorTok{-}\FloatTok{0.00275}\NormalTok{, }\FloatTok{0.0067}\NormalTok{),}
                     \DataTypeTok{breaks =} \KeywordTok{c}\NormalTok{(}\OperatorTok{-}\FloatTok{0.002}\NormalTok{, }\FloatTok{0.000}\NormalTok{, }\FloatTok{0.002}\NormalTok{, }\FloatTok{0.004}\NormalTok{),}
                     \DataTypeTok{labels =} \KeywordTok{c}\NormalTok{(}\StringTok{"-0.002"}\NormalTok{,}\StringTok{"0.000"}\NormalTok{,}\StringTok{"0.002"}\NormalTok{,}\StringTok{"0.004"}\NormalTok{),}
                     \DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{annotate}\NormalTok{(}\StringTok{"text"}\NormalTok{, }\DataTypeTok{x =}\NormalTok{ g2_all_bs}\OperatorTok{$}\NormalTok{g2, }\DataTypeTok{y =} \DecValTok{1340}\NormalTok{, }
           \DataTypeTok{label =} \KeywordTok{paste0}\NormalTok{(}\StringTok{'p = '}\NormalTok{, }\KeywordTok{round}\NormalTok{(g2_all_bs}\OperatorTok{$}\NormalTok{p, }\DecValTok{3}\NormalTok{)),}
           \DataTypeTok{family =} \KeywordTok{theme_get}\NormalTok{()}\OperatorTok{$}\NormalTok{text[[}\StringTok{"family"}\NormalTok{]],}
           \DataTypeTok{size =} \KeywordTok{theme_get}\NormalTok{()}\OperatorTok{$}\NormalTok{text[[}\StringTok{"size"}\NormalTok{]]}\OperatorTok{/}\DecValTok{4}\NormalTok{) }

\NormalTok{all_graphs_g2_immune_bs_histogram <-}
\StringTok{  }\NormalTok{ggplot2}\OperatorTok{::}\KeywordTok{ggplot}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme_classic}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_histogram}\NormalTok{(}\DataTypeTok{binwidth =} \FloatTok{0.00375}\NormalTok{, }\DataTypeTok{data =}\NormalTok{ g2_immune_bs, }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ bs),}
                 \DataTypeTok{color =} \StringTok{"#C1403D"}\NormalTok{,}
                 \DataTypeTok{fill =} \StringTok{"#C1403D"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_errorbarh}\NormalTok{(}\DataTypeTok{data =}\NormalTok{ g2_immune_bs,}
                 \KeywordTok{aes}\NormalTok{(}\DataTypeTok{y =} \DecValTok{1110}\NormalTok{, }\DataTypeTok{x =}\NormalTok{ g2, }\DataTypeTok{xmin =}\NormalTok{ lcl, }\DataTypeTok{xmax =}\NormalTok{ ucl),}
                 \DataTypeTok{color =} \StringTok{"black"}\NormalTok{, }\DataTypeTok{size =} \FloatTok{0.7}\NormalTok{, }\DataTypeTok{linetype =} \StringTok{"solid"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_linerange}\NormalTok{(}\DataTypeTok{data =}\NormalTok{ g2_immune_bs,  }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{ymin =} \DecValTok{0}\NormalTok{, }\DataTypeTok{ymax =} \DecValTok{1110}\NormalTok{, }\DataTypeTok{x =}\NormalTok{ g2),}
                 \DataTypeTok{linetype =} \StringTok{'dotted'}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme}\NormalTok{(}\DataTypeTok{text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{size =} \DecValTok{12}\NormalTok{),}
        \DataTypeTok{panel.border =} \KeywordTok{element_blank}\NormalTok{(),}
        \DataTypeTok{strip.background =} \KeywordTok{element_rect}\NormalTok{(}\DataTypeTok{fill =} \StringTok{"white"}\NormalTok{, }\DataTypeTok{colour =} \StringTok{"white"}\NormalTok{),}
        \DataTypeTok{strip.text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{colour =} \StringTok{'white'}\NormalTok{),}
        \DataTypeTok{plot.margin =}\NormalTok{ grid}\OperatorTok{::}\KeywordTok{unit}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{), }\StringTok{'mm'}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{facet_wrap}\NormalTok{(}\OperatorTok{~}\NormalTok{p) }\OperatorTok{+}
\StringTok{  }\KeywordTok{ylab}\NormalTok{(}\StringTok{" "}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{xlab}\NormalTok{(}\StringTok{" "}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_y_continuous}\NormalTok{(}\DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{), }\DataTypeTok{limits =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{1250}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_x_continuous}\NormalTok{(}\DataTypeTok{limits =} \KeywordTok{c}\NormalTok{(}\OperatorTok{-}\FloatTok{0.05}\NormalTok{, }\FloatTok{0.067}\NormalTok{),}
                     \DataTypeTok{breaks =} \KeywordTok{seq}\NormalTok{(}\OperatorTok{-}\FloatTok{0.05}\NormalTok{, }\FloatTok{0.05}\NormalTok{, }\FloatTok{0.05}\NormalTok{),}
                     \DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{annotate}\NormalTok{(}\StringTok{"text"}\NormalTok{, }\DataTypeTok{x =}\NormalTok{ g2_immune_bs}\OperatorTok{$}\NormalTok{g2, }\DataTypeTok{y =} \DecValTok{1139}\NormalTok{, }
           \DataTypeTok{label =} \KeywordTok{paste0}\NormalTok{(}\StringTok{'p = '}\NormalTok{, }\KeywordTok{round}\NormalTok{(g2_immune_bs}\OperatorTok{$}\NormalTok{p, }\DecValTok{3}\NormalTok{)),}
           \DataTypeTok{family =} \KeywordTok{theme_get}\NormalTok{()}\OperatorTok{$}\NormalTok{text[[}\StringTok{"family"}\NormalTok{]],}
           \DataTypeTok{size =} \KeywordTok{theme_get}\NormalTok{()}\OperatorTok{$}\NormalTok{text[[}\StringTok{"size"}\NormalTok{]]}\OperatorTok{/}\FloatTok{3.8}\NormalTok{) }
\end{Highlighting}
\end{Shaded}

\includegraphics{R-code_files/figure-latex/g2 all graphs-1.pdf}

\paragraph{(B) Calculate g2 for subsets of the
data}\label{b-calculate-g2-for-subsets-of-the-data}

Here, we repeat the estimation of g2 for each marker type and for the
entire dataset

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{g2_neutral_resampled <-}\StringTok{ }\NormalTok{pbapply}\OperatorTok{::}\KeywordTok{pblapply}\NormalTok{(}\KeywordTok{seq}\NormalTok{(}\DecValTok{4}\NormalTok{, }\DecValTok{48}\NormalTok{, }\DecValTok{4}\NormalTok{), }\ControlFlowTok{function}\NormalTok{(x) \{ }\OperatorTok{-}\NormalTok{->}
\StringTok{  }\NormalTok{subs <-}\StringTok{ }\KeywordTok{lapply}\NormalTok{(}\DecValTok{1}\OperatorTok{:}\DecValTok{100}\NormalTok{, }\ControlFlowTok{function}\NormalTok{(y) \{}
\NormalTok{    rand <-}\StringTok{ }\KeywordTok{sample}\NormalTok{(}\DecValTok{1}\OperatorTok{:}\DecValTok{48}\NormalTok{, x, }\DataTypeTok{replace =} \OtherTok{FALSE}\NormalTok{)}
\NormalTok{    loci <-}\StringTok{ }\NormalTok{neutral_markers[, rand]}
\NormalTok{    g2 <-}\StringTok{ }\KeywordTok{g2_microsats}\NormalTok{(loci, }\DataTypeTok{nperm =} \DecValTok{0}\NormalTok{, }\DataTypeTok{nboot =} \DecValTok{9999}\NormalTok{, }\DataTypeTok{verbose =}\NormalTok{ F)}
\NormalTok{    df <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{bs =}\NormalTok{ g2}\OperatorTok{$}\NormalTok{g2_boot,}
                     \DataTypeTok{lcl =}\NormalTok{ g2}\OperatorTok{$}\NormalTok{CI_boot[[}\DecValTok{1}\NormalTok{]],}
                     \DataTypeTok{ucl =}\NormalTok{ g2}\OperatorTok{$}\NormalTok{CI_boot[[}\DecValTok{2}\NormalTok{]],}
                     \DataTypeTok{g2  =}\NormalTok{ g2}\OperatorTok{$}\NormalTok{g2,}
                     \DataTypeTok{p =}\NormalTok{ g2}\OperatorTok{$}\NormalTok{p_val)}
    \KeywordTok{return}\NormalTok{(df[}\DecValTok{1}\NormalTok{,])}
\NormalTok{  \}) }\OperatorTok{%>%}\StringTok{ }\KeywordTok{do.call}\NormalTok{(}\StringTok{"rbind"}\NormalTok{, .)}
  \KeywordTok{return}\NormalTok{(}\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{g2 =} \KeywordTok{mean}\NormalTok{(subs}\OperatorTok{$}\NormalTok{g2),}
                    \DataTypeTok{lcl =} \KeywordTok{confidence_interval}\NormalTok{(subs}\OperatorTok{$}\NormalTok{g2)[}\DecValTok{1}\NormalTok{],}
                    \DataTypeTok{ucl =} \KeywordTok{confidence_interval}\NormalTok{(subs}\OperatorTok{$}\NormalTok{g2)[}\DecValTok{2}\NormalTok{]))}
\NormalTok{\}) }\OperatorTok{%>%}\StringTok{  }\KeywordTok{do.call}\NormalTok{(}\StringTok{"rbind"}\NormalTok{, .)}
\NormalTok{g2_neutral_resampled}\OperatorTok{$}\NormalTok{loci <-}\StringTok{ }\KeywordTok{seq}\NormalTok{(}\DecValTok{4}\NormalTok{, }\DecValTok{48}\NormalTok{, }\DecValTok{4}\NormalTok{)}
\KeywordTok{save}\NormalTok{(g2_neutral_resampled, }\DataTypeTok{file =} \StringTok{"data/g2_neutral_resampled.RData"}\NormalTok{)    }\CommentTok{# save the data}

\NormalTok{g2_neutral_resampled_plot <-}
\StringTok{  }\KeywordTok{ggplot}\NormalTok{(}\DataTypeTok{data =}\NormalTok{ g2_neutral_resampled, }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ loci, }\DataTypeTok{y =}\NormalTok{ g2)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_line}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_point}\NormalTok{(}\DataTypeTok{size =} \FloatTok{1.5}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_errorbar}\NormalTok{(}\KeywordTok{aes}\NormalTok{(}\DataTypeTok{ymin =}\NormalTok{ lcl,}
                    \DataTypeTok{ymax =}\NormalTok{ ucl),}
                \DataTypeTok{width =} \FloatTok{0.8}\NormalTok{, }\DataTypeTok{alpha =} \FloatTok{0.7}\NormalTok{, }\DataTypeTok{size =} \FloatTok{0.8}\NormalTok{, }\DataTypeTok{colour =} \StringTok{"black"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_hline}\NormalTok{(}\DataTypeTok{yintercept =} \DecValTok{0}\NormalTok{, }\DataTypeTok{linetype =} \StringTok{"dotted"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme_classic}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme}\NormalTok{(}\DataTypeTok{legend.position =} \StringTok{"none"}\NormalTok{,}
        \DataTypeTok{panel.border =} \KeywordTok{element_blank}\NormalTok{(),}
        \DataTypeTok{strip.background =} \KeywordTok{element_blank}\NormalTok{(),}
        \DataTypeTok{text =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{size =} \DecValTok{12}\NormalTok{),}
        \DataTypeTok{aspect.ratio =} \DecValTok{1}\NormalTok{,}
        \DataTypeTok{axis.title.y =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{face =} \StringTok{"italic"}\NormalTok{),}
        \DataTypeTok{plot.margin =}\NormalTok{ grid}\OperatorTok{::}\KeywordTok{unit}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{2}\NormalTok{), }\StringTok{'mm'}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{xlab}\NormalTok{(}\StringTok{"Number of loci"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{labs}\NormalTok{(}\DataTypeTok{y =} \KeywordTok{expression}\NormalTok{(}\KeywordTok{italic}\NormalTok{(g)[}\StringTok{"2"}\NormalTok{])) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_x_continuous}\NormalTok{(}\DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{), }\DataTypeTok{limits =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{, }\DecValTok{50}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\paragraph{(C) Heat map}\label{c-heat-map}

Next, we test for patterns in allelic richness among markers
(i.e.~immune vs neutral), developmental source (i.e.~designed for
Antarctic fur seals, phocids or otariids). Secondly, we evaluate the
cross-amplification success of loci in two other species of pinnipeds,
namely Grey seal and Nothern Elephant seal.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# Read and format genotypes}
\NormalTok{heatmap_df <-}\StringTok{ }\NormalTok{readxl}\OperatorTok{::}\KeywordTok{read_xlsx}\NormalTok{(}\StringTok{"data/genotypes_raw.xlsx"}\NormalTok{, }\DataTypeTok{skip =} \DecValTok{1}\NormalTok{)[, }\KeywordTok{c}\NormalTok{(}\DecValTok{3}\NormalTok{, }\DecValTok{8}\OperatorTok{:}\KeywordTok{ncol}\NormalTok{(seals))]}

\CommentTok{# Randomly select six individuals per species}
\NormalTok{heatmap_df <-}\StringTok{ }\NormalTok{heatmap_df[}\KeywordTok{c}\NormalTok{(}\KeywordTok{sample}\NormalTok{(}\KeywordTok{which}\NormalTok{(heatmap_df[[}\StringTok{"Species"}\NormalTok{]] }\OperatorTok{==}\StringTok{ "Fur seal"}\NormalTok{), }\DataTypeTok{size =} \DecValTok{6}\NormalTok{, }\DataTypeTok{replace =}\NormalTok{ F),}
           \KeywordTok{sample}\NormalTok{(}\KeywordTok{which}\NormalTok{(heatmap_df[[}\StringTok{"Species"}\NormalTok{]] }\OperatorTok{==}\StringTok{ "Grey seal"}\NormalTok{), }\DataTypeTok{size =} \DecValTok{6}\NormalTok{, }\DataTypeTok{replace =}\NormalTok{ F),}
           \KeywordTok{sample}\NormalTok{(}\KeywordTok{which}\NormalTok{(heatmap_df[[}\StringTok{"Species"}\NormalTok{]] }\OperatorTok{==}\StringTok{ "Northern Elephant seal"}\NormalTok{), }\DataTypeTok{size =} \DecValTok{6}\NormalTok{, }\DataTypeTok{replace =}\NormalTok{ F)),]}

\CommentTok{# Extract geno}
\NormalTok{marker_geno <-}\StringTok{ }\KeywordTok{apply}\NormalTok{(heatmap_df[,}\OperatorTok{-}\DecValTok{1}\NormalTok{], }\DecValTok{2}\NormalTok{, as.character)}

\CommentTok{# Get names of loci}
\NormalTok{loci_names <-}\StringTok{ }\KeywordTok{colnames}\NormalTok{(marker_geno)[}\KeywordTok{seq}\NormalTok{(}\DecValTok{1}\NormalTok{, }\KeywordTok{ncol}\NormalTok{(marker_geno), }\DecValTok{2}\NormalTok{)] }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{substring}\NormalTok{(., }\DataTypeTok{first =} \DecValTok{1}\NormalTok{, }\DataTypeTok{last =} \KeywordTok{nchar}\NormalTok{(.) }\OperatorTok{-}\StringTok{ }\DecValTok{2}\NormalTok{)}

\CommentTok{# Define a vector of Immune marker names}
\NormalTok{immune_marker_names <-}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"Agi01"}\NormalTok{, }\StringTok{"Agi02"}\NormalTok{, }\StringTok{"Agi03"}\NormalTok{, }\StringTok{"Agi04"}\NormalTok{,}
                         \StringTok{"Agi05"}\NormalTok{, }\StringTok{"Agi06"}\NormalTok{, }\StringTok{"Agi07"}\NormalTok{, }\StringTok{"Agi08"}\NormalTok{,}
                         \StringTok{"Agi09"}\NormalTok{, }\StringTok{"Agi10"}\NormalTok{, }\StringTok{"Agi11"}\NormalTok{, }\StringTok{"Agt10"}\NormalTok{, }\StringTok{"Agt47"}\NormalTok{)}

\CommentTok{# Collapse information for each locus in one column}
\NormalTok{marker_geno <-}\StringTok{ }\KeywordTok{lapply}\NormalTok{(}\KeywordTok{seq}\NormalTok{(}\DecValTok{1}\NormalTok{, }\KeywordTok{ncol}\NormalTok{(marker_geno), }\DecValTok{2}\NormalTok{), }\ControlFlowTok{function}\NormalTok{(x) \{}
\NormalTok{  marker_geno[,x}\OperatorTok{:}\NormalTok{(x }\OperatorTok{+}\StringTok{ }\DecValTok{1}\NormalTok{)] }\OperatorTok{%>%}\StringTok{ }
\StringTok{    }\KeywordTok{apply}\NormalTok{(., }\DecValTok{1}\NormalTok{, paste0, }\DataTypeTok{collapse =} \StringTok{"/"}\NormalTok{)}
\NormalTok{\}) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{do.call}\NormalTok{(}\StringTok{"cbind"}\NormalTok{,.) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\NormalTok{## rename loci}
\StringTok{  }\KeywordTok{set_colnames}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ ., }\DataTypeTok{value =} \KeywordTok{paste0}\NormalTok{(}\StringTok{"Locus"}\NormalTok{, }\DecValTok{1}\OperatorTok{:}\DecValTok{61}\NormalTok{))}

\NormalTok{## set missing data to NA}
\NormalTok{marker_geno[}\KeywordTok{which}\NormalTok{(marker_geno }\OperatorTok{==}\StringTok{ "NA/NA"}\NormalTok{)] <-}\StringTok{ }\OtherTok{NA}

\NormalTok{## convert to GENIND object}
\NormalTok{genind <-}\StringTok{ }\NormalTok{adegenet}\OperatorTok{::}\KeywordTok{df2genind}\NormalTok{(marker_geno, }\DataTypeTok{ploidy =} \DecValTok{2}\NormalTok{, }\DataTypeTok{sep =} \StringTok{"/"}\NormalTok{, }\DataTypeTok{pop =}\NormalTok{ heatmap_df[[}\StringTok{"Species"}\NormalTok{]] }\OperatorTok{%>%}\StringTok{ }\NormalTok{as.factor)}

\NormalTok{## Convert to GENPOP}
\NormalTok{genpop <-}\StringTok{ }\NormalTok{adegenet}\OperatorTok{::}\KeywordTok{genind2genpop}\NormalTok{(genind)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Converting data from a genind to a genpop object... 
## 
## ...done.
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{heatmap_df <-}\StringTok{ }\KeywordTok{lapply}\NormalTok{(}\KeywordTok{levels}\NormalTok{(genpop}\OperatorTok{@}\NormalTok{loc.fac), }\ControlFlowTok{function}\NormalTok{(i) \{}
\NormalTok{  df.temp <-}\StringTok{ }\NormalTok{genpop}\OperatorTok{@}\NormalTok{tab[,}\KeywordTok{which}\NormalTok{(genpop}\OperatorTok{@}\NormalTok{loc.fac }\OperatorTok{==}\StringTok{ }\NormalTok{i)]   ## fetch data }
  \ControlFlowTok{if}\NormalTok{ (}\KeywordTok{is.null}\NormalTok{(}\KeywordTok{dim}\NormalTok{(df.temp))) \{}
\NormalTok{    df.temp[df.temp }\OperatorTok{>}\StringTok{ }\DecValTok{0}\NormalTok{] <-}\StringTok{ }\DecValTok{1}
\NormalTok{    df.temp[df.temp }\OperatorTok{==}\StringTok{ }\DecValTok{0}\NormalTok{] <-}\StringTok{ }\DecValTok{0}
    
\NormalTok{  \} }\ControlFlowTok{else}\NormalTok{ \{}
\NormalTok{    df.temp <-}\StringTok{   }\KeywordTok{apply}\NormalTok{(df.temp, }\DecValTok{2}\NormalTok{, }\ControlFlowTok{function}\NormalTok{(x) }\KeywordTok{ifelse}\NormalTok{(x }\OperatorTok{>}\StringTok{ }\DecValTok{0}\NormalTok{, }\DecValTok{1}\NormalTok{, }\DecValTok{0}\NormalTok{)) }\OperatorTok{%>%}\StringTok{ }\NormalTok{## presence/absence of allele}
\StringTok{      }\KeywordTok{rowSums}\NormalTok{(}\DataTypeTok{na.rm =}\NormalTok{ T) ## count alleles}
\NormalTok{  \}}
  \CommentTok{# return results}
  \KeywordTok{return}\NormalTok{(}\KeywordTok{data.frame}\NormalTok{(}\DataTypeTok{Species =} \KeywordTok{names}\NormalTok{(df.temp),}
                    \DataTypeTok{Locus =}\NormalTok{ i,}
                    \DataTypeTok{Alleles =}\NormalTok{ df.temp))}
\NormalTok{\}) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{do.call}\NormalTok{(}\StringTok{"rbind"}\NormalTok{, .)}

\NormalTok{## set zero to NA}
\NormalTok{heatmap_df[[}\StringTok{"Alleles"}\NormalTok{]][}\KeywordTok{which}\NormalTok{(heatmap_df[[}\StringTok{"Alleles"}\NormalTok{]] }\OperatorTok{==}\StringTok{ }\DecValTok{0}\NormalTok{)] <-}\StringTok{ }\OtherTok{NA}

\NormalTok{heatmap_df[[}\StringTok{"Locus"}\NormalTok{]] <-}\StringTok{ }\KeywordTok{factor}\NormalTok{(heatmap_df[[}\StringTok{"Locus"}\NormalTok{]], }\DataTypeTok{labels =}\NormalTok{ loci_names)}
\NormalTok{heatmap_df[[}\StringTok{"Type"}\NormalTok{]] <-}\StringTok{ 'Neutral'}
\NormalTok{heatmap_df[[}\StringTok{"Type"}\NormalTok{]][}\KeywordTok{which}\NormalTok{(heatmap_df[[}\StringTok{"Locus"}\NormalTok{]] }\OperatorTok{%in%}\StringTok{ }\NormalTok{immune_marker_names)] <-}\StringTok{ 'Immune'}

\NormalTok{## sort by species}
\NormalTok{heatmap_df[[}\StringTok{"Species"}\NormalTok{]] <-}\StringTok{ }\KeywordTok{factor}\NormalTok{(heatmap_df[[}\StringTok{"Species"}\NormalTok{]],}
                          \DataTypeTok{levels =} \KeywordTok{c}\NormalTok{(}\StringTok{"Fur seal"}\NormalTok{, }\StringTok{"Grey seal"}\NormalTok{, }\StringTok{"Northern Elephant seal"}\NormalTok{),}
                          \DataTypeTok{labels =} \KeywordTok{c}\NormalTok{(}\StringTok{"Antarctic fur seal"}\NormalTok{, }\StringTok{"Grey seal"}\NormalTok{, }\StringTok{"Northern Elephant seal"}\NormalTok{))}

\NormalTok{## define colours for marker types}
\NormalTok{col_key <-}\StringTok{ }\KeywordTok{ifelse}\NormalTok{(}\KeywordTok{levels}\NormalTok{(heatmap_df[[}\StringTok{"Locus"}\NormalTok{]]) }\OperatorTok{%in%}\StringTok{ }\NormalTok{immune_marker_names, }\StringTok{"#C1403D"}\NormalTok{, }\StringTok{"#0294A5"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{plot <-}\StringTok{ }\KeywordTok{ggplot}\NormalTok{(}\DataTypeTok{data =}\NormalTok{ heatmap_df, }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ Species, }\DataTypeTok{y =}\NormalTok{ Locus, }\DataTypeTok{fill =}\NormalTok{ Alleles)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme_classic}\NormalTok{() }\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_tile}\NormalTok{(}\DataTypeTok{colour =} \StringTok{"Black"}\NormalTok{, }\DataTypeTok{size =}\NormalTok{ .}\DecValTok{75}\NormalTok{) }\OperatorTok{+}\StringTok{ }
\StringTok{  }\KeywordTok{scale_fill_viridis_c}\NormalTok{(}\DataTypeTok{name =} \StringTok{"Alleles"}\NormalTok{, }\DataTypeTok{na.value =} \StringTok{"Grey75"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{scale_x_discrete}\NormalTok{(}\DataTypeTok{expand =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{)) }\OperatorTok{+}
\StringTok{  }\KeywordTok{theme}\NormalTok{(}
    \DataTypeTok{plot.margin =} \KeywordTok{margin}\NormalTok{(}\DataTypeTok{t =} \DecValTok{5}\NormalTok{, }\DataTypeTok{r =} \DecValTok{25}\NormalTok{, }\DataTypeTok{b =} \DecValTok{5}\NormalTok{, }\DataTypeTok{l =} \DecValTok{15}\NormalTok{, }\DataTypeTok{unit =} \StringTok{"mm"}\NormalTok{),}
    \DataTypeTok{legend.position =} \KeywordTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{,}\DecValTok{1}\NormalTok{), }
    \DataTypeTok{legend.justification =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{, }\DecValTok{1}\NormalTok{),}
    \DataTypeTok{legend.direction =} \StringTok{"vertical"}\NormalTok{,}
    \DataTypeTok{legend.margin =} \KeywordTok{margin}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{,}\DecValTok{5}\NormalTok{, }\StringTok{"mm"}\NormalTok{),}
    \DataTypeTok{axis.text.y =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{hjust =} \DecValTok{0}\NormalTok{, }\DataTypeTok{colour =}\NormalTok{ col_key, }\DataTypeTok{size =} \DecValTok{10}\NormalTok{),}
    \DataTypeTok{axis.line.x =} \KeywordTok{element_blank}\NormalTok{(),}
    \DataTypeTok{axis.text.x =} \KeywordTok{element_text}\NormalTok{(}\DataTypeTok{size =} \DecValTok{10}\NormalTok{)) }\OperatorTok{+}\StringTok{ }
\StringTok{  }\KeywordTok{xlab}\NormalTok{(}\StringTok{"Species"}\NormalTok{) }\OperatorTok{+}
\StringTok{  }\KeywordTok{ylab}\NormalTok{(}\StringTok{""}\NormalTok{) }
\NormalTok{plot}
\end{Highlighting}
\end{Shaded}

\includegraphics{R-code_files/figure-latex/unnamed-chunk-10-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{ggsave}\NormalTok{(plot,}
       \DataTypeTok{filename =} \StringTok{'HeatmapLoci.tiff'}\NormalTok{,}
       \DataTypeTok{width =} \DecValTok{6}\NormalTok{,}
       \DataTypeTok{height =} \DecValTok{9}\NormalTok{,}
       \DataTypeTok{units =} \StringTok{"in"}\NormalTok{,}
       \DataTypeTok{dpi =} \DecValTok{300}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Compare allelic richness}\label{compare-allelic-richness}

The heatmap above shows several patterns which are tested statistically
next.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{## get raw data again}
\NormalTok{genotypes_raw <-}\StringTok{ }\NormalTok{readxl}\OperatorTok{::}\KeywordTok{read_xlsx}\NormalTok{(}\StringTok{"data/genotypes_raw.xlsx"}\NormalTok{, }\DataTypeTok{skip =} \DecValTok{1}\NormalTok{)[, }\KeywordTok{c}\NormalTok{(}\DecValTok{3}\NormalTok{, }\DecValTok{8}\OperatorTok{:}\KeywordTok{ncol}\NormalTok{(seals))]}

\CommentTok{# Extract genotypes}
\NormalTok{marker_geno <-}\StringTok{ }\KeywordTok{apply}\NormalTok{(genotypes_raw[,}\OperatorTok{-}\DecValTok{1}\NormalTok{], }\DecValTok{2}\NormalTok{, as.character)}

\CommentTok{# Collapse information for each locus in one column}
\NormalTok{marker_geno <-}\StringTok{ }\KeywordTok{lapply}\NormalTok{(}\KeywordTok{seq}\NormalTok{(}\DecValTok{1}\NormalTok{, }\KeywordTok{ncol}\NormalTok{(marker_geno), }\DecValTok{2}\NormalTok{), }\ControlFlowTok{function}\NormalTok{(x) \{}
\NormalTok{  marker_geno[,x}\OperatorTok{:}\NormalTok{(x }\OperatorTok{+}\StringTok{ }\DecValTok{1}\NormalTok{)] }\OperatorTok{%>%}\StringTok{ }
\StringTok{    }\KeywordTok{apply}\NormalTok{(., }\DecValTok{1}\NormalTok{, paste0, }\DataTypeTok{collapse =} \StringTok{"/"}\NormalTok{)}
\NormalTok{\}) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{do.call}\NormalTok{(}\StringTok{"cbind"}\NormalTok{,.) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\NormalTok{## rename loci}
\StringTok{  }\KeywordTok{set_colnames}\NormalTok{(}\DataTypeTok{x =}\NormalTok{ ., }\DataTypeTok{value =} \KeywordTok{paste0}\NormalTok{(}\StringTok{"Locus"}\NormalTok{, }\DecValTok{1}\OperatorTok{:}\DecValTok{61}\NormalTok{))}

\NormalTok{## set missing data to NA}
\NormalTok{marker_geno[}\KeywordTok{which}\NormalTok{(marker_geno }\OperatorTok{==}\StringTok{ "NA/NA"}\NormalTok{)] <-}\StringTok{ }\OtherTok{NA}

\NormalTok{## Create GENIND for Antarctic fur seal alone}
\NormalTok{genind_afs <-}\StringTok{ }\NormalTok{adegenet}\OperatorTok{::}\KeywordTok{df2genind}\NormalTok{(marker_geno[}\DecValTok{1}\OperatorTok{:}\DecValTok{78}\NormalTok{,], }\DataTypeTok{ploidy =} \DecValTok{2}\NormalTok{, }\DataTypeTok{sep =} \StringTok{"/"}\NormalTok{)}

\NormalTok{## extract allele numbers for both marker types}
\NormalTok{immune_afs <-}\StringTok{ }\NormalTok{genind}\OperatorTok{@}\NormalTok{loc.n.all[}\DecValTok{1}\OperatorTok{:}\DecValTok{13}\NormalTok{]}
\KeywordTok{mean}\NormalTok{(immune_afs)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4.153846
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(immune_afs)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2.478109
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{neutral_afs <-}\StringTok{ }\NormalTok{genind}\OperatorTok{@}\NormalTok{loc.n.all[}\DecValTok{14}\OperatorTok{:}\DecValTok{61}\NormalTok{]}
\KeywordTok{mean}\NormalTok{(neutral_afs)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 7.270833
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(neutral_afs)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2.490318
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{## compare marker types}
\KeywordTok{wilcox.test}\NormalTok{(immune_afs, neutral_afs, }\DataTypeTok{paired =}\NormalTok{ F)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  immune_afs and neutral_afs
## W = 116.5, p-value = 0.0005435
## alternative hypothesis: true location shift is not equal to 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{## compare neutral markers by origin}
\NormalTok{neutral_afs <-}\StringTok{ }\NormalTok{genind}\OperatorTok{@}\NormalTok{loc.n.all[}\DecValTok{14}\OperatorTok{:}\DecValTok{22}\NormalTok{]}
\KeywordTok{mean}\NormalTok{(neutral_afs)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4.666667
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(neutral_afs)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 1.802776
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{neutral_others <-}\StringTok{ }\NormalTok{genind}\OperatorTok{@}\NormalTok{loc.n.all[}\DecValTok{23}\OperatorTok{:}\DecValTok{61}\NormalTok{]}
\KeywordTok{mean}\NormalTok{(neutral_others)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 7.871795
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(neutral_others)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2.238179
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{## compare by marker}
\KeywordTok{wilcox.test}\NormalTok{(neutral_afs, neutral_others, }\DataTypeTok{paired =}\NormalTok{ F)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  neutral_afs and neutral_others
## W = 48.5, p-value = 0.0007452
## alternative hypothesis: true location shift is not equal to 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{## cross-amplification}
\NormalTok{immune <-}\StringTok{ }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{filter}\NormalTok{(heatmap_df, Species }\OperatorTok{!=}\StringTok{ "Antarctic fur seal"}\NormalTok{, Type }\OperatorTok{==}\StringTok{ "Immune"}\NormalTok{)[[}\StringTok{"Alleles"}\NormalTok{]]}
\NormalTok{immune <-}\StringTok{ }\KeywordTok{ifelse}\NormalTok{(}\KeywordTok{is.na}\NormalTok{(immune), }\DecValTok{0}\NormalTok{, }\DecValTok{1}\NormalTok{) }\CommentTok{# check if amplified}
\KeywordTok{mean}\NormalTok{(immune) ## cross-amplification rate}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0.6923077
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{neutral <-}\StringTok{ }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{filter}\NormalTok{(heatmap_df, Species }\OperatorTok{!=}\StringTok{ "Antarctic fur seal"}\NormalTok{)[}\DecValTok{27}\OperatorTok{:}\DecValTok{44}\NormalTok{, }\StringTok{"Alleles"}\NormalTok{]}
\NormalTok{neutral <-}\StringTok{ }\KeywordTok{ifelse}\NormalTok{(}\KeywordTok{is.na}\NormalTok{(neutral), }\DecValTok{0}\NormalTok{, }\DecValTok{1}\NormalTok{) }\CommentTok{# check if amplified }
\KeywordTok{mean}\NormalTok{(neutral) ## cross-amplification rate}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0.2222222
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{wilcox.test}\NormalTok{(neutral, immune, }\DataTypeTok{paired =}\NormalTok{ F)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  neutral and immune
## W = 124, p-value = 0.00255
## alternative hypothesis: true location shift is not equal to 0
\end{verbatim}


\end{document}
